File: HW1_Task3/src/com/company/Suit.java
package com.company;

public enum Suit {
    CLUBS(1), DIAMONDS(2), SPADES(3), HEARTS(4);
    private final int suitPower;

    Suit(int power) {
        suitPower = power;
    }

    public int getSuitValue() {
        return this.suitPower;
    }
}

File: HW3_Task1/HW3_Task1/src/Suit.kt
enum class Suit(suitValue: Int, suitName: String) {
    CLUBS(1, "clubs"), DIAMONDS(2, "diamonds"), SPADES(3, "spades"), HEARTS(4, "hearts");

    private val value = suitValue
    private val suit = suitName

    fun getSuitValue(): Int {
        return value
    }

    fun getSuitName(): String {
        return suit
    }
}